### PR TITLE
allow user-specified cmake flags

### PR DESCRIPTION
**Pull request by snickerbockers**

### CODE REVIEW

File: entrypoint [hunk 1]
```diff
@@ -24,6 +24,11 @@ if [ -n "$INPUT_CTESTFLAGS" ]; then
     ctest+=( "$INPUT_CTESTFLAGS" )
 fi
 
+cmake=( cmake )
+if [ -n "$INPUT_CMAKEFLAGS" ]; then
+    cmake+=( "$INPUT_CMAKEFLAGS" )
+fi
+
```
Comment (lpenz) [inline entrypoint:31]: I've been planning on doing this, thanks!
Have you checked if it works with multiple flags?

File: entrypoint [hunk 2]
```diff
@@ -33,7 +38,7 @@ fi
 if [ -n "$INPUT_ANALYZER" ]; then
     if [ "$INPUT_ANALYZER" = "clang-tidy" ]; then
         install_dependencies
-        cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON .
+        "${cmake[@]}" . -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
```
Comment (lpenz) [inline entrypoint:41]: I'm not so sure about this one... Which use case are you targeting?

File: entrypoint (hunk 1 above)
Comment (snickerbockers) [reply entrypoint]: :thinking: Looks like it wasn't, the quotes around `INPUT_CMAKEFLAGS` were getting propagated to the cmake command and that was causing it to ignore all of them after the first one.  I pushed out a new one that fixes this by removing the quotes.

[I tested it out here on one of my repos, and it's working toggling both options in the same CMakeLists.txt](https://github.com/washingtondc-emu/washingtondc/blob/github_actions_invariants/.github/workflows/build_linux.yml).

File: entrypoint (hunk 2 above)
Comment (snickerbockers) [reply entrypoint]: I was actually just copying that over for the sake of conformity, I don't actually have a use-case for that.  I deleted it in the latest push.

File: entrypoint [hunk 3]
```diff
@@ -24,6 +24,11 @@ if [ -n "$INPUT_CTESTFLAGS" ]; then
     ctest+=( "$INPUT_CTESTFLAGS" )
 fi
 
+cmake=( cmake )
+if [ -n "$INPUT_CMAKEFLAGS" ]; then
+    cmake+=( $INPUT_CMAKEFLAGS )
+fi
```
Comment (lpenz) [inline entrypoint:30]: I did some digging, can you try the following?

    IFS=' ' read -r -a cmakeflags <<< "$INPUT_CMAKEFLAGS"
    cmake+=( "${cmakeflags[@]" )

This creates a proper `cmakeflags` arrays, and then concatenates to the `cmake` array

Comment (snickerbockers) [reply entrypoint]: I just implemented that in the latest push, it works as expected.

**Comment by lpenz** [comment]
Merged, thanks once again for doing this

**Pull request merged**